### DISCUSSION TITLE
Insert dummy data using CLI

### DISCUSSION THREAD

**Discussion by oachoor** [directus/directus#11197] [Archive]
Hello,
We're using docker to run Directus 9.4.x
In the _command_ section we aim to insert a required dummy data somewhere after applying the schema, that's why we're not using bootstrap command.

`docker-compose.yaml`
```bash
volumes:
      - "./directus/schema/snapshot.yaml:/directus/snapshot.yaml:ro"
      - "./directus/extensions/migrations:/directus/extensions/migrations:ro"
command:
      - /bin/sh
      - -c
      - |
        npx directus database install
        npx directus users create --email foo --password bar --role `npx directus roles create --role Administrator --admin`
        npx directus schema apply --yes ./snapshot.yaml
        npx directus database migrate:latest
        npx directus database migrate:down
        npx directus database migrate:up
        npx directus start
```

The following error occurs only when using `database install` instead of `bootstrap` since the migrations are applied before the snapshot:
```
22:12:27 🚨 error select "collection", "singleton", "note", "sort_field", "accountability" from "directus_collections" - column "accountability" does not exist
error: select "collection", "singleton", "note", "sort_field", "accountability" from "directus_collections" - column "accountability" does not exist
    at Parser.parseErrorMessage (/directus/node_modules/pg-protocol/dist/parser.js:287:98)
    at Parser.handlePacket (/directus/node_modules/pg-protocol/dist/parser.js:126:29)
    at Parser.parse (/directus/node_modules/pg-protocol/dist/parser.js:39:38)
    at Socket.<anonymous> (/directus/node_modules/pg-protocol/dist/index.js:11:42)
    at Socket.emit (node:events:390:28)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:289:9)
    at Socket.Readable.push (node:internal/streams/readable:228:10)
    at TCP.onStreamRead (node:internal/stream_base_commons:199:23)
```

I this the correct approach to achieve that?

**Comment by joselcvarela** [comment]
I think you can replace this command 
```bash
npx directus users create --email foo --password bar --role `npx directus roles create --role Administrator --admin`
```
by using
```env
ADMIN_EMAIL=foo
ADMIN_PASSWORD=bar
```

Although, you should be able what you are tying to do, but I believe you need to change the order of things
```bash
        npx directus database install
        npx directus schema apply --yes ./snapshot.yaml
        npx directus database migrate:latest
        npx directus database migrate:down
        npx directus database migrate:up
        #this line comes here
        npx directus users create --email foo --password bar --role `npx directus roles create --role Administrator --admin` 
        npx directus start
```

Can you try that?

**Reply by oachoor** [reply]
Thanks for your quick reply,
Are you sure about by simply defining the `ADMIN_EMAIL`; `ADMIN_PASSWORD` variables the user/role get created? I think it's only covered by bootstrap command or?
Unfortunately the error persists with the order you suggested.
The thing is that the snapshot requires at least that system migrations are executed, to solve the "accountability" error, at least [Add Collections Accountability](https://github.com/directus/directus/blob/75b5f33727bff8f086d6f8e89f9eb2a09630a7cd/api/src/database/migrations/20210416A-add-collections-accountability.ts) migration should be carried out.

**Reply by oachoor** [reply]
It seems that migration is not suitable in that scenario, the question is, how to insert dummy data after applying the snapshot? 🤔

**Reply by joselcvarela** [reply]
So, you have this:
```
        npx directus database install
        npx directus users create --email foo --password bar --role `npx directus roles create --role Administrator --admin`
        npx directus schema apply --yes ./snapshot.yaml
        npx directus database migrate:latest
        npx directus database migrate:down
        npx directus database migrate:up
        npx directus start
```

Where are you inserting dummy data?
Also, why are you migrating to latest, rollback and go to latest again?

Can't you run just this? This should be the way to go.
```
npx directus bootstrap
npx directus schema apply --yes ./snapshot.yaml
npx directus users create --email foo --password bar --role `npx directus roles create --role Administrator --admin`
npx directus start
```

**Reply by oachoor** [reply]
For now the dummy data is part of the latest migration (sort of hacky), and it requires the snapshot to be applied first, that's why the down/up commands in order to re-run the latest migration.
When using `bootstrap` command, all migrations will be added before the schema is applied, which is not what I want.

**Comment by rijkvanzanten** [comment]
Ref https://github.com/directus/directus/discussions/10996
